internal/roles/roles.go

```diff
@@ -22,6 +22,9 @@ func IsDelegatedTargetsRole(name string) bool {
 }
 
 func IsTopLevelManifest(name string) bool {
+	if IsVersionedManifest(name) {
+		name = strings.Join(strings.Split(name, ".")[1:], ".")
```

Comment:
Maybe slightly safer when splitting by '.' to get only the last 2 tokens, but the function below will catch shenanigans anyway.

Reply:
[all nitpicks]

Note to self: it should never be possible that `name` has no `.` in it, because `IsVersionedManifest` checks that. Maybe worth an assertion (via `panic`)? (There's currently an implicit assertion when we index into the result of `Split` 😄 )

IMO this is nicer as `name := strings.SplitN(name, ".", 2)[1]` or `_, name, _ := strings.Cut(name, ".")`—WDYT?

Reply:
Nice! I forgot about `SplitN` and `Cut`.

The `Cut` solution is really nice because I can assert that it was found